oak_restricted_kernel_wrapper/src/elf.rs

```diff
@@ -68,7 +68,14 @@ fn load_segment(
     // Zero out the target in case the file content is shorter than the target.
     target.fill(0);
 
-    target[..file_length].copy_from_slice(source);
+    // Manually copy between slices to avoid the compiler's intrinsic memcpy which uses an indirect
```

Comment:
I'm surprised the compiler doesn't catch this. (Or maybe it does and inlines something more clever than just a stupid loop?)

Reply:
I haven't looked at the resulting assembly, but it seems likely that the compiler should be able to optimise this. This approach does bypass the core slice implementation that relies on the compiler intrinsics, which in turn leads to relocations in the ELF file.